### PR TITLE
Introduce VRML format (.wrl and .x3dv) 3D model support

### LINKED ISSUE #5862: Support loading VRML97 (.wrl) and "Classic VRML" (.x3dv) 3D model files
X3D is a superset of both VRML97 (`.wrl`) and "Classic VRML" (`.x3dv`) files.  The main difference is in the syntax/format of the files; X3D uses `.xml` while the others are based on the older VRML format.

Since the contents all three formats are understood by X3D, by simply converting the VRML syntax to `.xml`, both `.wrl` and `x3dv` files should be able to be loaded by the current `X3DImporter` class.

### PR DESCRIPTION
**Pull request by tellypresence**
Closes #5862 

### Overview
Introduce support for loading `.wrl` and `.x3dv` 3D model files.

X3D is a superset of the WRL/X3DV formats.  X3D migrated to `xml` file format, while WRL uses `VRML97` format and X3DV uses "classic VRML" format.  X3D is able to fully understand the content of the older files, on the condition that the formatting is converted from "classic VRML"/`VRML97` to `.xml` format.  Once converted, the file contents as `.xml` are able to be used with the extant X3D importer to load WRL/X3DV files.

### Technical details
_Summary_
Use a mature 3rd party tool to convert WRL/X3DV files on-the-fly to X3D (xml), then simply hand over to the X3D importer as usual.

_Details_
Leveraging the `Scanner` and `Parser` from the mature 3rd party [`meshlab`](https://github.com/cnr-isti-vclab/meshlab/tree/main/src/meshlabplugins/io_x3d/vrml) which provide the ability to load a `.wrl` (`VRML97`) or `.x3dv` ("classic VRML") file and convert on-the-fly from the respective old format to an X3D `.xml` file.

`X3DImporter` is modified to first check whether input file extension is either of `wrl` or `x3dv`, and in this case convert to X3D `.xml` via `Parser`, which provides an in-memory X3D xml file, which can then be handed to a new template override of the `XMLParser::parse()` accepting `std::istream`.

From that point, the common X3D importer code flow is followed regardless of whether the model originated as a native X3D `.xml` file or was converted on the fly from `VRML97` (`.wrl`) or "classic VRML" (`.x3dv`) format.

### Screenshots
The result seems to be satisfactory, basic mesh geometry for fairly complex models appear to be dealt with reasonably correctly.  Some models found "in the wild" have correct normals when imported, some end up with incorrect normals.

#### Models with correct normals

<table>
<tr><th rowspan=2>Model</th><th colspan=2>Screenshots</th><th rowspan=2>Notes</th></tr>
<tr><th>Diffuse lighting</th><th>Debug normals<br>X = Red, Y = Green, Z = Blue</th></tr>
<tr><td><a href="https://github.com/assimp/assimp/blob/master/test/models/WRL/Wuson.wrl"><tt>test/models/WRL/Wuson.wrl</tt></a></td><td>Wuson_wrl</td><td>Wuson_wrl_debug_normals</td><td></td></tr>
<tr><td><a href="https://www.artec3d.com/3d-models/borderlands-cosplayer"><tt>Borderlands cosplayer</tt></a></td><td>Borderlands cosplay-wrlBorderlands cosplay-wrl</td><td>Borderlands cosplay-wrl_debug_normalsBorderlands cosplay-wrl_debug_normals</td><td></td></tr>
<tr><td><a href="https://www.artec3d.com/3d-models/harley-quinn-cosplayer"><tt>Harley Quinn cosplayer</tt></a></td><td>Harley Quinn-wrlHarley Quinn-wrl</td><td>Harley Quinn-wrl_debug_normalsHarley Quinn-wrl_debug_normals</td><td></td></tr>
<tr><td><a href="https://www.cgtrader.com/free-3d-print-models/art/scans-replicas/3dprint-statue-of-a-sailor"><tt>statue of a sailor</tt></a></td><td>cgtrader_sailor_w_binoculars_wrl</td><td>cgtrader_sailor_w_binoculars_wrl_debug_normals_ok</td><td></td></tr>

</table>

#### Models with incorrect normals
<table>
<tr><th rowspan=2>Model</th><th colspan=2>Screenshots</th><th rowspan=2>Notes</th></tr>
<tr><th>Diffuse lighting</th><th>Debug normals<br>X = Red, Y = Green, Z = Blue</th></tr>
<tr><td><a href="https://github.com/assimp/assimp/blob/master/test/models/WRL/MotionCaptureROM.WRL"><tt>test/models/WRL/MotionCaptureROM.WRL</tt></a></td><td>MotionCaptureROM_wrl</td><td>MotionCaptureROM_wrl_debug_normals</td><td></td></tr>
<tr><td><a href="https://grabcad.com/library/brabham-bt62-2019-1"><tt>BRABHAM BT62 2019</tt></a></td><td>brabham-bt62-2019_wrl</td><td>brabham-bt62-2019_wrl_debug_normals</td><td></td></tr>
<tr><td><a href="https://downloadfree3d.com/3d-models/vehicles/car/mclaren-senna-2019/"><tt>McLaren SENNA 2019</tt></a></td><td>mclaren-senna-2019_wrlmclaren-senna-2019_wrl</td><td>mclaren-senna-2019_wrl_debug_normalsmclaren-senna-2019_wrl_debug_normals</td><td></td></tr>

</table>

### Caveats
- Note that the X3D importer is currently in a broken state, as detailed in
  - #4992
   Since the WRL/X3DV model file is simply convered to an X3D `.xml` file, the result will only be as good as the `X3DImporter`.

- Normals are correct for some, and incorrect for other `.wrl` models tested, but probably better to get this feature into the hands of users and investigate problems later (could be the models themselves have bad normals)

### Issues related to this PR
This PR addresses one of the formats requested in #3145

### CODE REVIEW

**Comment by tellypresence** [comment]
@kimkulling Please review when you have time, this PR provides the ability to load `.wrl` files which wasn't possible previously.

**Comment by tellypresence** [comment]
Seems like most of the Sonar Cloud notices are related to code copied from the 3rd party external `meshlab` project, or advocating for C++17 features (e.g. replacing `const std::string&` w/`std::string_view`), wondering if it will be okay to ignore those and merge this PR anyway

**Comment by tellypresence** [comment]
@kimkulling Any thoughts on the Sonar Cloud issues outlined above?

**Review by kimkulling — CHANGES_REQUESTED** [review]

File: code/AssetLib/VRML/VrmlConverter.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+/// \brief  Convert VRML-formatted (.wrl, .x3dv) files to X3D .xml format
+/// \date   2024
+/// \author tellypresence
+
+#ifndef ASSIMP_BUILD_NO_X3D_IMPORTER
+
+#include <memory> // std::unique_ptr
+#include "VrmlConverter.hpp"
+
+using std::string;
```
[the remark this answers is not in the document]
Comment (tellypresence) [reply code/AssetLib/VRML/VrmlConverter.cpp]: Removed "using" decl

File: code/AssetLib/X3D/X3DImporter.cpp [hunk 2]
```diff
@@ -45,6 +45,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #ifndef ASSIMP_BUILD_NO_X3D_IMPORTER
 
+#include "../VRML/VrmlConverter.hpp"
```
[the remark this answers is not in the document]
Comment (tellypresence) [reply code/AssetLib/X3D/X3DImporter.cpp]: Replaced relative path

File: code/AssetLib/X3D/X3DImporter.cpp [hunk 3]
```diff
@@ -215,7 +224,18 @@ void X3DImporter::ParseFile(const std::string &file, IOSystem *pIOHandler) {
     if (!theParser.parse(fileStream.get())) {
         return;
     }
+    ParseFile(theParser);
+}
+
+void X3DImporter::ParseFile(std::istream &myIstream) {
+    XmlParser theParser;
+    if (!theParser.parse(myIstream)) {
+        return;
```
[the remark this answers is not in the document]
Comment (tellypresence) [reply code/AssetLib/X3D/X3DImporter.cpp:234]: Added log statement

**Review by kimkulling — CHANGES_REQUESTED** [review]
Take a look into the sonarcube hint: use std::string_view instead of std::string.

**Comment by tellypresence** [comment]
> Take a look into the sonarcube hint: use std::string_view instead of std::string.

`std::string_view` is a C++17 feature, is it OK to use it?  Here's a line in [`FBXConverter.cpp`](https://github.com/assimp/assimp/blob/7bd54e360450d2fbff5fe7e3a97ccfe6fdb04570/code/AssetLib/FBX/FBXConverter.cpp#L221) which notes that `std::string_view` would be an improvement but continues to use `std::string` to avoid causing problems for users with older build systems

**Comment by kimkulling** [comment]
I will merge this PR and take care of the sonarcube findings. Great work!

**Pull request merged**